postgres/datadog_checks/postgres/config.py

```diff
@@ -2,322 +2,566 @@
+
+    return tags, errors
+
+
+METRIC_TYPES = {
+    'RATE': AgentCheck.rate,
+    'GAUGE': AgentCheck.gauge,
+    'MONOTONIC': AgentCheck.monotonic_count,
+}
+
+
+def map_custom_metrics(custom_metrics):
+    # Pre-process custom metrics and verify definition
+    required_parameters = ("descriptors", "metrics", "query", "relation")
+
+    for m in custom_metrics:
+        for param in required_parameters:
+            if param not in m:
+                raise ConfigurationError('Missing {} parameter in custom metric'.format(param))
+
+        # Old formatting to new formatting. The first params is always the columns names from which to
+        # read metrics. The `relation` param instructs the check to replace the next '%s' with the list of
+        # relations names.
+        if m['relation']:
+            m['query'] = m['query'] % ('{metrics_columns}', '{relations_names}')
         else:
-            # if managed_authentication is set, we ignore the legacy managed_identity config
-            managed_authentication = azure['managed_authentication']
-            enabled = is_affirmative(managed_authentication.get('enabled', False))
-            if enabled and 'client_id' not in managed_authentication:
-                raise ConfigurationError('Azure client_id must be set when using Azure managed authentication')
-            managed_authentication['enabled'] = enabled
-        return managed_authentication
-
-    @staticmethod
-    def _should_collect_wal_metrics(collect_wal_metrics) -> Optional[bool]:
-        if collect_wal_metrics is not None:
-            # if the user has explicitly set the value, return the boolean
-            return is_affirmative(collect_wal_metrics)
-
-        return None
-
-    @staticmethod
-    def _should_propagate_agent_tags(instance, init_config) -> bool:
-        '''
-        return True if the agent tags should be propagated to the check
-        '''
-        instance_propagate_agent_tags = instance.get('propagate_agent_tags')
-        init_config_propagate_agent_tags = init_config.get('propagate_agent_tags')
-
-        if instance_propagate_agent_tags is not None:
-            # if the instance has explicitly set the value, return the boolean
-            return instance_propagate_agent_tags
-        if init_config_propagate_agent_tags is not None:
-            # if the init_config has explicitly set the value, return the boolean
-            return init_config_propagate_agent_tags
-        # if neither the instance nor the init_config has set the value, return False
-        return False
+            m['query'] = m['query'] % '{metrics_columns}'
+
+        try:
+            for ref, (_, mtype) in m['metrics'].items():
+                cap_mtype = mtype.upper()
+                if cap_mtype not in ('RATE', 'GAUGE', 'MONOTONIC'):
+                    raise ConfigurationError(
+                        'Collector method {} is not known. Known methods are RATE, GAUGE, MONOTONIC'.format(cap_mtype)
```

Comment:
Similarly here. I think it would be helpful to know which custom metric has an invalid type